### PR TITLE
Fix official+unofficial submit

### PR DESCRIPTION
**Pull request by icanberk**

### CODE REVIEW

**Comment by icanberk** [comment]
Fixes issue #126

File: app/models/submission.rb [hunk 1]
```diff
@@ -91,6 +91,8 @@ def saveFile(upload)
       # Sanity!
       upload['file'].rewind
       File.open(path,"wb") { |f| f.write(upload['file'].read)}
+    elsif upload['local_submit_file']
+      File.open(path,"wb") { |f| f.write(IO.read(upload['local_submit_file']))}
```
Comment (dlbucci) [inline app/models/submission.rb]: Why are there two different methods for reading the file here?

File: app/controllers/assessments_controller.rb [hunk 2]
```diff
@@ -1299,10 +1299,16 @@ def local_submit
         @cud = CourseUserDatum.find_cud_for_course(@course, @user.id)
         @submission = Submission.create(:assessment_id=>@assessment.id,
           :course_user_datum_id => @cud.id);
-        @submission.filename = File.join(internalDir, handinFile)
-        @submission.save!
+        upload = {'local_submit_file'=>File.join(internalDir, handinFile)}
 
-        afterHandin(@submission)
+        @submission.saveFile(upload)
+        #afterHandin(@submission)
```
Comment (dlbucci) [inline app/controllers/assessments_controller.rb]: Dead code here.

File: app/models/submission.rb (hunk 1 above)
Comment (icanberk) [inline app/models/submission.rb]: upload['file'] is a class that we're not using in the case of local submit. so there is not way to do `.read`

**Comment by dlbucci** [comment]
If we're gonna say this fixes #126, this should fix the scripts/autolabSubmit.sh too.

**Comment by icanberk** [comment]
Is 213 using that script? Otherwise, can we just get rid of it?

File: app/models/submission.rb (hunk 1 above)
Comment (dlbucci) [inline app/models/submission.rb]: So upload['file'] is an actually file-like object, but upload['local_submit_file'] is a string?  Could we get a comment for that.

**Comment by dlbucci** [comment]
122 and 210 are using that script for command line submission.  We should have an official local submit script.

**Comment by namanbharadwaj** [comment]
We (210) actually have our own script that follows the protocol, so as long as the protocol is documented, getting rid of the official script is perfectly fine. I believe that 122 does a similar thing (@robsimmons).

**Pull request merged**